### PR TITLE
Ticket #1075: Generate current-full.csv and current-federal.csv daily and publish at a public URL

### LINKED ISSUE #1075: Generate current-full.csv and current-federal.csv daily and publish at a public URL
### Issue Description

#996 prepared 3 different reports and made them available as on-demand CSV exports. For this issue, we should provide the capability to generate 2 of those reports – current-full.csv and current-federal.csv – and have them accessible through an API.

### Acceptance criteria

- [ ] Save the files (delete/overwrite any prior copies) and make them available at a public URL
- [ ] Cameron can use the public URL to replace the `http_url` value in our [Git scraping instance](https://github.com/cisagov/dotgov-data/blob/main/.github/workflows/flat.yml).

### Issue Links

⛔ Blocked by #996
Dependency of #1403, #1317 (That ticket is for notification for when this process fails to complete properly.)

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1075 

## Changes

- Adds daily-csv-upload.yaml which generates current-full.csv and current-federal.csv. The cron job is currently commented out until #1403 is done so we don't continually generate empty files.
- Adds two new api endpoints in views.py for get_current_full and get_current_federal + helper function
- Modifies urls.py to use the new api endpoints
- Adds the command generate_current_federal_report.py
- Adds the command generate_current_full_report.py
- Add test cases

## Context for reviewers
Some portions of this PR have "TODO" and are not fully implemented. The reason for this is that to fully persist these files on our sandboxes, we need a S3 AWS instance. This is being handled in  #1403.

After this is implemented, some code will have to change with how the file is being grabbed / pushed, and the portion of the yaml which uses the cron job will be readded. For now, we can test the core functionality + architecture of this ticket.

I am currently making the assumption that we want to return a FileResponse for the flat library to read it, but if that is incorrect it can be changed to a HttpResponse. 

## Setup
Since this ticket concerns our csv exports, you will need to import some data to accurately test this ticket. Navigate to src, then download [this file ](https://drive.google.com/file/d/195DjlREITUBJB64tIcjchNBbHikY3dmk/view?usp=drive_link) and extract the file content into the `migrationdata/` folder within the project. Then, run these two commands locally:

```
docker-compose exec app ./manage.py load_transition_domain migrationFilepaths.json --limitParse 5000
docker-compose exec app ./manage.py transfer_transition_domains_to_domains
```

Now, with data loaded, you can test the added functionality of this ticket. While this will be automated, for now the way to test this to run these manually. Run these two commands:

```
docker-compose exec app ./manage.py generate_current_federal_report
docker-compose exec app ./manage.py generate_current_full_report
```
Both of these should output "Success!". These files are generated under the migrationdata/ folder, so you should see two .csv files and they should be populated as you would otherwise expect.

Afterwards, go to your browser and navigate to these two urls:

 localhost:8080/api/v1/get-report/current-full
 localhost:8080/api/v1/get-report/current-federal
  
After navigating to these, your browser should automagically download the associated csv files, and all information therein should be correct. This is so that the [dotgov-data Action](https://github.com/cisagov/dotgov-data/blob/main/.github/workflows/flat.yml) can reach out to this url and get this information.  

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: .github/workflows/daily-csv-upload.yaml
Comment (zandercymatics) [inline .github/workflows/daily-csv-upload.yaml:42]: "za" will eventually be changed to "stable". It is set as za for now to avoid breaking things if a bug causes some issue

File: .github/workflows/daily-csv-upload.yaml
Comment (rachidatecs) [inline .github/workflows/daily-csv-upload.yaml:53]: I believe you'll need to revise the script used per @neilmb's PR. Again on line 62.

**Review by erinysong — APPROVED** [review]
Checked the following could be successfully performed locally after downloading data on PR:
- [x] Load and transfer transition domains to domains using `load_transition_domain` and `transfer_transition_domains_to_domains`
- [x] Generate local csv files for current federal report and current full report
- [x] Access a downloaded version of the CSV files using the urls /api/v1/get-report/current-full and /api/v1/get-report/current-federal. The CSV files appear to have the same data as the csv files generated locally.

**Review by erinysong — COMMENTED** [review]
Ack sorry had the files changed window open while testing so did not see @rachidatecs 
's comment catching the script changes to match @neilmb's updates. Do not think I saw anything that stood out when reviewing the rest of the code and did verify that the changes work locally on my end, so feel free to request re-review if need be

File: src/api/views.py
Comment (abroddrick) [inline src/api/views.py:96]: Add comment about what this file should contain, a dev looking at this a year from now would have to do some detective work to understand the purpose and what this file is

File: src/api/views.py
Comment (abroddrick) [inline src/api/views.py:102]: same comment as above

**Review by abroddrick — COMMENTED** [review]
Based on your description it sounds like this doesn't actually meet the full AC of 1075.  I know we talked about modify the ac of 1075 and moving the relevant portions out, into a follow on ticket (what would have been 1403). Was there a reason to keep 1075 dependent on 1403? As is, I don't think the approvals are valid with the AC still staying as is. Won't you need another approval after 1403 is merged in?

**Comment by zandercymatics** [comment]
@abroddrick Thanks for pointing that out, I corrected the AC

File: .github/workflows/daily-csv-upload.yaml
Comment (zandercymatics) [reply .github/workflows/daily-csv-upload.yaml:53]: Thanks Rachid, I assume to "cloud-gov/cg-cli-tools@main"?

**Pull request closed without merging**
